### ISSUE TITLE
With PIA being acquired by Kape, support for other VPN providers?

### ISSUE THREAD

**Issue by elmerfdz** [qdm12/private-internet-access-docker#59]
With Kape acquiring PIA, will you ever consider building an image for a different provider like Windscribe, the only reason I haven't switched yet is because I really like this image and there aren't any good or many Windscribe images on DockerHub.

Thanks

**Comment by qdm12** [comment]
It's in process, I'm changing the shell scripting to a Go compiled static binary to handle the init and all the services running in the container (openvpn, unbound, shadowsocks, tinyproxy, etc etc) and to allow for easier maintainability/testing. Once that's done, I'll add VPN providers and I'll likely need help to test it out! The first one on the list is [Mullvad](https://mullvad.net/en/); I guess Windscribe will be second! 👍

**Comment by elmerfdz** [comment]
That's fantastic news, glad you'll be supporting other providers. I did narrow down my search to Mullvad & Windscribe but the 3 year BF deal on Windscribe was too cheap to pass. Let me know when you've added support, happy to help you out with testing. 😀

**Comment by burkasaurusrex** [comment]
Love this container and glad to hear you're planning to support other providers. Lower priority but it would be cool to have the option to use Wireguard as well instead of OpenVPN (pretty sure both Mullvad and Windscribe support it)

**Comment by qdm12** [comment]
Yes that's also part of the plan. I'm working on another Docker image with a userspace Wireguard program (not in the kernel). Probably a tiny bit slower, but easily Dockerized. Once it's done I'll add it to this image!

**Comment by groenator** [comment]
> It's in process, I'm changing the shell scripting to a Go compiled static binary to handle the init and all the services running in the container (openvpn, unbound, shadowsocks, tinyproxy, etc etc) and to allow for easier maintainability/testing. Once that's done, I'll add VPN providers and I'll likely need help to test it out! The first one on the list is [Mullvad](https://mullvad.net/en/); I guess Windscribe will be second! 👍

If you need help on testing the image, let me know. I am a windscribe provider,  I would be very interested in testing this image with different providers.

**Comment by qdm12** [comment]
I'll get back to you as soon as these are being added.

I am still working on the Go branch, re-implementing the shell scripting with some added goodies. You can track latest commits [here](https://github.com/qdm12/private-internet-access-docker/commits/go) and will bundle mullvad and windscribe support before merging the branch, so I'll need your testing :+1:

**Comment by groenator** [comment]
Awesome! Thanks for including me for testing. I will wait for your update.

Regards,

**Comment by qdm12** [comment]
The Go rewrite Docker image is now out in 'beta' at `qmcgaw/private-internet-access:go`.

- The documentation is [here](https://github.com/qdm12/private-internet-access-docker/tree/go#private-internet-access-client)
- It should work with the same settings as before
- Feel free to comment here or on the [Slack channel](https://join.slack.com/t/qdm12/shared_invite/enQtOTE0NjcxNTM1ODc5LTYyZmVlOTM3MGI4ZWU0YmJkMjUxNmQ4ODQ2OTAwYzMxMTlhY2Q1MWQyOWUyNjc2ODliNjFjMDUxNWNmNzk5MDk)
- Note that it's lacking port forwarding for now, I will do it in the coming few days
- It might have bugs as it's not totally unit tested yet

I will add more providers soon once this Go image is ready for prime time!

**Comment by groenator** [comment]
Great! Thanks for the update, I will give it a try this weekend.

Regards,

**Comment by Frepke** [comment]
I'm testing the go branch for now, so far no noticeable problems.
Installation went without problems.

**Comment by ljmerza** [comment]
Will this repo be moved to another name once other VPNs are added?

**Comment by qdm12** [comment]
It will likely stay with the same repository and docker image name I think. Although I'm opened to suggestions renaming the Github repository, or adding another Docker image name mirroring the current one.

By the way I started development for Mullvad this morning, for now I'm mapping all their countries, cities and IP addresses :smile: it is tiring haha, but it should be relatively quick to implement overall for openvpn.

**Comment by ljmerza** [comment]
I was thinking something like `simple-vpn` or similar since it automates port forwarding, kill switch, etc. I think people are moving away from the PIA brand so people seeing the name on the repo might overlook this awesome tool.

**Comment by elmerfdz** [comment]
Agree with @ljmerza, it might be best to move away from the PIA branding once you've added support for more providers. Can't think of a good name at the moment 😐. 

I'll get it added to the Unraid community applications once you come to a decision on that.

**Comment by qdm12** [comment]
Maybe *gluetun* - eases your tunneling by gluing pieces together for you ™️  😄

**Comment by ljmerza** [comment]
hey, it works

**Comment by qdm12** [comment]
Hello all, I'm working on adding support for Mullvad. Can someone please try it?

- [Documentation](https://github.com/qdm12/private-internet-access-docker/tree/mullvad#environment-variables) (especially new env variables, don't forget to set `VPNSP=mullvad`)
- Docker image is `qmcgaw/private-internet-access:mullvad` (only for amd64 for testing)

You can also try it for PIA and it should work as before (`VPNSP=pia`)

**Comment by qdm12** [comment]
There is now a potentially working Docker image for Mullvad `qmcgaw/private-internet-access:mullvad` built for all architectures (ARM etc.), feel free to try it (also has a bunch of improvements for PIA as well). I'll merge it next week and start working on Windscribe.

**Comment by qdm12** [comment]
I merged the Mullvad support in the latest tag, I'll start development for Windscribe next week and ping back here once there is a beta version.

**Comment by qdm12** [comment]
Hello, for people using Mullvad (I think @ljmerza if my memory is right) is it stable enough in your opinion? I would like to do another release + docker tag with support for pia and mullvad only before we move on to Windscribe.

**Comment by qdm12** [comment]
I'm trying to access [openvpn config files for Windscribe](https://windscribe.com/getconfig/openvpn) but I can't, can anyone subscribed with them attach some of their Openvpn configs (ideally all if it's possible)? Thanks!

**Comment by Frepke** [comment]
> I'm trying to access [openvpn config files for Windscribe](https://windscribe.com/getconfig/openvpn) but I can't, can anyone subscribed with them attach some of their Openvpn configs (ideally all if it's possible)? Thanks!

Here you go
[Windscribe-OVPN-TCP.zip](https://github.com/qdm12/private-internet-access-docker/files/4366468/Windscribe-OVPN-TCP.zip)
[Windscribe-OVPN-UDP.zip](https://github.com/qdm12/private-internet-access-docker/files/4366470/Windscribe-OVPN-UDP.zip)

**Comment by qdm12** [comment]
Thanks @Frepke 

I made all the changes for Windscribe [on this branch](https://github.com/qdm12/private-internet-access-docker/tree/windscribe), however, from the files attached, I only added the regions I found there (see [this code](https://github.com/qdm12/private-internet-access-docker/blob/windscribe/internal/constants/windscribe.go#L23-L150)) although there seem to be more regions/cities from [https://windscribe.com/status](https://windscribe.com/status). Could anyone subscribed with Windscribe send the updated openvpn files (careful to remove your username/password if there is any).

Thanks!

**Comment by elmerfdz** [comment]
Here you go mate:

[Windscribe-OVPN-UDP.zip](https://github.com/qdm12/private-internet-access-docker/files/4372719/Windscribe-OVPN-UDP.zip)
[Windscribe-OVPN-TCP.zip](https://github.com/qdm12/private-internet-access-docker/files/4372720/Windscribe-OVPN-TCP.zip)

**Comment by Frepke** [comment]
> 
> 
> Thanks @Frepke
> 
> I made all the changes for Windscribe [on this branch](https://github.com/qdm12/private-internet-access-docker/tree/windscribe), however, from the files attached, I only added the regions I found there (see [this code](https://github.com/qdm12/private-internet-access-docker/blob/windscribe/internal/constants/windscribe.go#L23-L150)) although there seem to be more regions/cities from https://windscribe.com/status. Could anyone subscribed with Windscribe send the updated openvpn files (careful to remove your username/password if there is any).
> 
> Thanks!
> 
> EDIT: YOu can try the image with the tag `:windscribe` and using the environment variable `REGION=netherlands`, and set your user and password as before.

Hmmm...

I think :windscribe isn't a tag for now.

**Comment by qdm12** [comment]
Indeed, the build failed because of this silly `s390x` cpu architecture no one uses 😄 ... Anyway it should be building now and will create a `:windscribe` tag (I'm also about to move to Github actions for building and more visibility).

I also added all the missing regions so it should be good for beta testing 😉

**Comment by Frepke** [comment]
> 
> 
> Indeed, the build failed because of this silly `s390x` cpu architecture no one uses 😄 ... Anyway it should be building now and will create a `:windscribe` tag (I'm also about to move to Github actions for building and more visibility).
> 
> I also added all the missing regions so it should be good for beta testing 😉

Waiting for the new build and then...
...testing 😄

**Comment by Frepke** [comment]
1st test: VPNSP can't be windscribe, it has to be pia or mullvad.

**Comment by Frepke** [comment]
2nd test: won't run
`
2020-03-24T14:33:59.824+0100	INFO	openvpn configurator: checking for device /dev/net/tun


2020-03-24T14:33:59.825+0100	WARN	TUN device is not available: open /dev/net/tun: no such file or directory


2020-03-24T14:33:59.825+0100	INFO	openvpn configurator: creating /dev/net/tun


2020-03-24T14:33:59.825+0100	INFO	openvpn configurator: writing auth file /etc/openvpn/auth.conf


2020-03-24T14:33:59.827+0100	INFO	firewall configurator: accepting all traffic


2020-03-24T14:33:59.839+0100	INFO	dns configurator: using DNS address 1.1.1.1 internally


2020-03-24T14:33:59.840+0100	INFO	dns configurator: downloading root hints from https://raw.githubusercontent.com/qdm12/files/master/named.root.updated


2020-03-24T14:33:59.841+0100	INFO	Launching standard output merger


2020-03-24T14:34:00.395+0100	INFO	dns configurator: downloading root key from https://raw.githubusercontent.com/qdm12/files/master/root.key.updated


2020-03-24T14:34:00.576+0100	INFO	dns configurator: generating Unbound configuration


2020-03-24T14:34:01.550+0100	INFO	dns configurator: 131467 hostnames blocked overall


2020-03-24T14:34:01.551+0100	INFO	dns configurator: 9 IP addresses blocked overall


2020-03-24T14:34:01.901+0100	INFO	dns configurator: starting unbound


2020-03-24T14:34:01.903+0100	INFO	dns configurator: using DNS address 127.0.0.1 internally


2020-03-24T14:34:01.904+0100	INFO	dns configurator: using DNS address 127.0.0.1 system wide


2020-03-24T14:34:01.910+0100	WARN	could not resolve github.com (try 1 of 10)


2020-03-24T14:34:03.383+0100	INFO	unbound: [1585056843] unbound[22:0] notice: init module 0: validator


2020-03-24T14:34:03.383+0100	INFO	unbound: [1585056843] unbound[22:0] notice: init module 1: iterator


2020-03-24T14:34:03.422+0100	INFO	unbound: [1585056843] unbound[22:0] info: start of service (unbound 1.9.6).


2020-03-24T14:34:03.453+0100	INFO	unbound: [1585056843] unbound[22:0] info: generate keytag query _ta-4a5c-4f66. NULL IN


2020-03-24T14:34:03.653+0100	INFO	firewall configurator: detecting default network route


2020-03-24T14:34:03.654+0100	INFO	firewall configurator: default route found: interface eth0, gateway 172.17.0.1, subnet 172.17.0.0/16


2020-03-24T14:34:03.657+0100	INFO	firewall configurator: adding 10.54.1.0/24 as route via eth0


2020-03-24T14:34:03.660+0100	INFO	firewall configurator: clearing all rules


2020-03-24T14:34:03.670+0100	INFO	firewall configurator: blocking all traffic


2020-03-24T14:34:03.682+0100	INFO	firewall configurator: creating general rules


2020-03-24T14:34:03.702+0100	INFO	firewall configurator: allowing output traffic to VPN server 185.212.171.131 through eth0 on port udp 443


2020-03-24T14:34:03.707+0100	INFO	firewall configurator: allowing output traffic to VPN server 84.17.46.2 through eth0 on port udp 443


2020-03-24T14:34:03.711+0100	INFO	firewall configurator: allowing output traffic to VPN server 72.11.157.35 through eth0 on port udp 443


2020-03-24T14:34:03.715+0100	INFO	firewall configurator: allowing output traffic to VPN server 72.11.157.67 through eth0 on port udp 443


2020-03-24T14:34:03.719+0100	INFO	firewall configurator: allowing output traffic to VPN server 185.253.96.3 through eth0 on port udp 443


2020-03-24T14:34:03.725+0100	INFO	firewall configurator: allowing output traffic to VPN server 37.120.192.19 through eth0 on port udp 443


2020-03-24T14:34:03.729+0100	INFO	firewall configurator: allowing output traffic to VPN server 46.166.143.98 through eth0 on port udp 443


2020-03-24T14:34:03.738+0100	INFO	firewall configurator: accepting input and output traffic for 172.17.0.0/16


2020-03-24T14:34:03.744+0100	INFO	firewall configurator: accepting input traffic through eth0 from 10.54.1.0/24 to 172.17.0.0/16


2020-03-24T14:34:03.746+0100	INFO	firewall configurator: accepting output traffic through eth0 from 172.17.0.0/16 to 10.54.1.0/24


2020-03-24T14:34:03.749+0100	INFO	shadowsocks configurator: generating configuration file


2020-03-24T14:34:03.749+0100	INFO	firewall configurator: accepting any input traffic on port 8388


2020-03-24T14:34:03.757+0100	INFO	shadowsocks configurator: starting shadowsocks server


2020-03-24T14:34:03.759+0100	INFO	openvpn configurator: starting openvpn


2020-03-24T14:34:03.763+0100	INFO	shadowsocks:  2020-03-24 14:34:03 INFO: UDP relay enabled


2020-03-24T14:34:03.763+0100	INFO	shadowsocks:  2020-03-24 14:34:03 INFO: initializing ciphers... chacha20-ietf-poly1305


2020-03-24T14:34:03.764+0100	INFO	shadowsocks:  2020-03-24 14:34:03 INFO: using nameserver: 127.0.0.1


2020-03-24T14:34:03.764+0100	INFO	shadowsocks:  2020-03-24 14:34:03 INFO: tcp server listening at 0.0.0.0:8388


2020-03-24T14:34:03.764+0100	INFO	shadowsocks:  2020-03-24 14:34:03 INFO: udp server listening at 0.0.0.0:8388


2020-03-24T14:34:03.767+0100	INFO	openvpn: Tue Mar 24 14:34:03 2020 OpenVPN 2.4.8 x86_64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on Feb  7 2020


2020-03-24T14:34:03.767+0100	INFO	openvpn: Tue Mar 24 14:34:03 2020 library versions: OpenSSL 1.1.1d  10 Sep 2019, LZO 2.10


2020-03-24T14:34:03.774+0100	INFO	openvpn: Tue Mar 24 14:34:03 2020 TCP/UDP: Preserving recently used remote address: [AF_INET]185.212.171.131:443


2020-03-24T14:34:03.774+0100	INFO	openvpn: Tue Mar 24 14:34:03 2020 UDP link local: (not bound)


2020-03-24T14:34:03.775+0100	INFO	openvpn: Tue Mar 24 14:34:03 2020 UDP link remote: [AF_INET]185.212.171.131:443


2020-03-24T14:34:03.775+0100	INFO	openvpn: Tue Mar 24 14:34:03 2020 NOTE: UID/GID downgrade will be delayed because of --client, --pull, or --up-delay


2020-03-24T14:35:03.803+0100	INFO	openvpn: Tue Mar 24 14:35:03 2020 [UNDEF] Inactivity timeout (--ping-restart), restarting


2020-03-24T14:35:03.803+0100	INFO	openvpn: Tue Mar 24 14:35:03 2020 SIGUSR1[soft,ping-restart] received, process restarting


2020-03-24T14:35:13.804+0100	INFO	openvpn: Tue Mar 24 14:35:13 2020 TCP/UDP: Preserving recently used remote address: [AF_INET]185.253.96.3:443


2020-03-24T14:35:13.804+0100	INFO	openvpn: Tue Mar 24 14:35:13 2020 UDP link local: (not bound)


2020-03-24T14:35:13.804+0100	INFO	openvpn: Tue Mar 24 14:35:13 2020 UDP link remote: [AF_INET]185.253.96.3:443


2020-03-24T14:36:13.512+0100	INFO	openvpn: Tue Mar 24 14:36:13 2020 [UNDEF] Inactivity timeout (--ping-restart), restarting


2020-03-24T14:36:13.512+0100	INFO	openvpn: Tue Mar 24 14:36:13 2020 SIGUSR1[soft,ping-restart] received, process restarting


2020-03-24T14:36:23.512+0100	INFO	openvpn: Tue Mar 24 14:36:23 2020 TCP/UDP: Preserving recently used remote address: [AF_INET]46.166.143.98:443


2020-03-24T14:36:23.513+0100	INFO	openvpn: Tue Mar 24 14:36:23 2020 UDP link local: (not bound)


2020-03-24T14:36:23.513+0100	INFO	openvpn: Tue Mar 24 14:36:23 2020 UDP link remote: [AF_INET]46.166.143.98:443


2020-03-24T14:37:23.518+0100	INFO	openvpn: Tue Mar 24 14:37:23 2020 [UNDEF] Inactivity timeout (--ping-restart), restarting


2020-03-24T14:37:23.518+0100	INFO	openvpn: Tue Mar 24 14:37:23 2020 SIGUSR1[soft,ping-restart] received, process restarting


2020-03-24T14:37:33.519+0100	INFO	openvpn: Tue Mar 24 14:37:33 2020 TCP/UDP: Preserving recently used remote address: [AF_INET]37.120.192.19:443


2020-03-24T14:37:33.519+0100	INFO	openvpn: Tue Mar 24 14:37:33 2020 UDP link local: (not bound)


2020-03-24T14:37:33.519+0100	INFO	openvpn: Tue Mar 24 14:37:33 2020 UDP link remote: [AF_INET]37.120.192.19:443


2020-03-24T14:38:33.979+0100	INFO	openvpn: Tue Mar 24 14:38:33 2020 [UNDEF] Inactivity timeout (--ping-restart), restarting


2020-03-24T14:38:33.979+0100	INFO	openvpn: Tue Mar 24 14:38:33 2020 SIGUSR1[soft,ping-restart] received, process restarting


2020-03-24T14:38:43.980+0100	INFO	openvpn: Tue Mar 24 14:38:43 2020 TCP/UDP: Preserving recently used remote address: [AF_INET]72.11.157.67:443


2020-03-24T14:38:43.981+0100	INFO	openvpn: Tue Mar 24 14:38:43 2020 UDP link local: (not bound)


2020-03-24T14:38:43.981+0100	INFO	openvpn: Tue Mar 24 14:38:43 2020 UDP link remote: [AF_INET]72.11.157.67:443


2020-03-24T14:39:43.186+0100	INFO	openvpn: Tue Mar 24 14:39:43 2020 [UNDEF] Inactivity timeout (--ping-restart), restarting


2020-03-24T14:39:43.186+0100	INFO	openvpn: Tue Mar 24 14:39:43 2020 SIGUSR1[soft,ping-restart] received, process restarting


2020-03-24T14:39:53.187+0100	INFO	openvpn: Tue Mar 24 14:39:53 2020 TCP/UDP: Preserving recently used remote address: [AF_INET]72.11.157.35:443


2020-03-24T14:39:53.187+0100	INFO	openvpn: Tue Mar 24 14:39:53 2020 UDP link local: (not bound)


2020-03-24T14:39:53.187+0100	INFO	openvpn: Tue Mar 24 14:39:53 2020 UDP link remote: [AF_INET]72.11.157.35:443`

**Comment by Frepke** [comment]
I thought by inserting a code, the comment wouldn't be that large :(

**Comment by qdm12** [comment]
You need to surround your code with 3 backticks :wink:

Ok so the problem comes from the openvpn config and Windscribe not being happy with it:

```
[UNDEF] Inactivity timeout (--ping-restart), restarting
```

I'll fix it at noon (in 2 hours).

**Comment by Frepke** [comment]
Still no new image

**Comment by Frepke** [comment]
3rd test: no luck

```

2020-03-24T21:18:19.685+0100	INFO	openvpn configurator: checking for device /dev/net/tun

2020-03-24T21:18:19.685+0100	WARN	TUN device is not available: open /dev/net/tun: no such file or directory

2020-03-24T21:18:19.685+0100	INFO	openvpn configurator: creating /dev/net/tun

2020-03-24T21:18:19.685+0100	INFO	openvpn configurator: writing auth file /etc/openvpn/auth.conf

2020-03-24T21:18:19.686+0100	INFO	firewall configurator: accepting all traffic

2020-03-24T21:18:19.697+0100	INFO	Launching standard output merger

2020-03-24T21:18:19.697+0100	INFO	dns configurator: using DNS address 1.1.1.1 internally

2020-03-24T21:18:19.697+0100	INFO	dns configurator: downloading root hints from https://raw.githubusercontent.com/qdm12/files/master/named.root.updated

2020-03-24T21:18:19.980+0100	INFO	dns configurator: downloading root key from https://raw.githubusercontent.com/qdm12/files/master/root.key.updated

2020-03-24T21:18:19.990+0100	INFO	dns configurator: generating Unbound configuration

2020-03-24T21:18:21.010+0100	INFO	dns configurator: 131467 hostnames blocked overall

2020-03-24T21:18:21.011+0100	INFO	dns configurator: 9 IP addresses blocked overall

2020-03-24T21:18:21.351+0100	INFO	dns configurator: starting unbound

2020-03-24T21:18:21.353+0100	INFO	dns configurator: using DNS address 127.0.0.1 internally

2020-03-24T21:18:21.354+0100	INFO	dns configurator: using DNS address 127.0.0.1 system wide

2020-03-24T21:18:21.359+0100	WARN	could not resolve github.com (try 1 of 10)

2020-03-24T21:18:22.737+0100	INFO	unbound: [1585081102] unbound[21:0] notice: init module 0: validator

2020-03-24T21:18:22.737+0100	INFO	unbound: [1585081102] unbound[21:0] notice: init module 1: iterator

2020-03-24T21:18:22.777+0100	INFO	unbound: [1585081102] unbound[21:0] info: start of service (unbound 1.9.6).

2020-03-24T21:18:22.809+0100	INFO	unbound: [1585081102] unbound[21:0] info: generate keytag query _ta-4a5c-4f66. NULL IN

2020-03-24T21:18:23.028+0100	INFO	firewall configurator: detecting default network route

2020-03-24T21:18:23.028+0100	INFO	firewall configurator: default route found: interface eth0, gateway 172.17.0.1, subnet 172.17.0.0/16

2020-03-24T21:18:23.031+0100	INFO	firewall configurator: adding 10.54.1.0/24 as route via eth0

2020-03-24T21:18:23.034+0100	INFO	firewall configurator: clearing all rules

2020-03-24T21:18:23.043+0100	INFO	firewall configurator: blocking all traffic

2020-03-24T21:18:23.052+0100	INFO	firewall configurator: creating general rules

2020-03-24T21:18:23.066+0100	INFO	firewall configurator: allowing output traffic to VPN server 185.212.171.131 through eth0 on port udp 443

2020-03-24T21:18:23.071+0100	INFO	firewall configurator: allowing output traffic to VPN server 84.17.46.2 through eth0 on port udp 443

2020-03-24T21:18:23.074+0100	INFO	firewall configurator: allowing output traffic to VPN server 72.11.157.35 through eth0 on port udp 443

2020-03-24T21:18:23.077+0100	INFO	firewall configurator: allowing output traffic to VPN server 46.166.143.98 through eth0 on port udp 443

2020-03-24T21:18:23.080+0100	INFO	firewall configurator: allowing output traffic to VPN server 185.253.96.3 through eth0 on port udp 443

2020-03-24T21:18:23.083+0100	INFO	firewall configurator: allowing output traffic to VPN server 72.11.157.67 through eth0 on port udp 443

2020-03-24T21:18:23.086+0100	INFO	firewall configurator: allowing output traffic to VPN server 37.120.192.19 through eth0 on port udp 443

2020-03-24T21:18:23.093+0100	INFO	firewall configurator: accepting input and output traffic for 172.17.0.0/16

2020-03-24T21:18:23.098+0100	INFO	firewall configurator: accepting input traffic through eth0 from 10.54.1.0/24 to 172.17.0.0/16

2020-03-24T21:18:23.101+0100	INFO	firewall configurator: accepting output traffic through eth0 from 172.17.0.0/16 to 10.54.1.0/24

2020-03-24T21:18:23.103+0100	INFO	shadowsocks configurator: generating configuration file

2020-03-24T21:18:23.104+0100	INFO	firewall configurator: accepting any input traffic on port 8388

2020-03-24T21:18:23.112+0100	INFO	shadowsocks configurator: starting shadowsocks server

2020-03-24T21:18:23.113+0100	INFO	openvpn configurator: starting openvpn

2020-03-24T21:18:23.120+0100	INFO	shadowsocks:  2020-03-24 21:18:23 INFO: UDP relay enabled

2020-03-24T21:18:23.120+0100	INFO	shadowsocks:  2020-03-24 21:18:23 INFO: initializing ciphers... chacha20-ietf-poly1305

2020-03-24T21:18:23.121+0100	INFO	openvpn: Tue Mar 24 21:18:23 2020 OpenVPN 2.4.8 x86_64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on Feb  7 2020

2020-03-24T21:18:23.121+0100	INFO	shadowsocks:  2020-03-24 21:18:23 INFO: using nameserver: 127.0.0.1

2020-03-24T21:18:23.121+0100	INFO	openvpn: Tue Mar 24 21:18:23 2020 library versions: OpenSSL 1.1.1d  10 Sep 2019, LZO 2.10

2020-03-24T21:18:23.121+0100	INFO	shadowsocks:  2020-03-24 21:18:23 INFO: tcp server listening at 0.0.0.0:8388

2020-03-24T21:18:23.121+0100	INFO	shadowsocks:  2020-03-24 21:18:23 INFO: udp server listening at 0.0.0.0:8388

2020-03-24T21:18:23.125+0100	INFO	openvpn: Tue Mar 24 21:18:23 2020 TCP/UDP: Preserving recently used remote address: [AF_INET]37.120.192.19:443

2020-03-24T21:18:23.125+0100	INFO	openvpn: Tue Mar 24 21:18:23 2020 UDP link local: (not bound)

2020-03-24T21:18:23.126+0100	INFO	openvpn: Tue Mar 24 21:18:23 2020 UDP link remote: [AF_INET]37.120.192.19:443

2020-03-24T21:18:23.126+0100	INFO	openvpn: Tue Mar 24 21:18:23 2020 NOTE: UID/GID downgrade will be delayed because of --client, --pull, or --up-delay

2020-03-24T21:19:12.822+0100	INFO	shadowsocks:  2020-03-24 21:19:12 INFO: new connection from client, 1 opened client connections

2020-03-24T21:19:12.823+0100	INFO	shadowsocks:  2020-03-24 21:19:12 INFO: [8388] connect to bolt.dropbox.com:443

2020-03-24T21:19:14.811+0100	INFO	shadowsocks:  2020-03-24 21:19:14 INFO: new connection from client, 2 opened client connections

2020-03-24T21:19:14.811+0100	INFO	shadowsocks:  2020-03-24 21:19:14 INFO: [8388] connect to p28-caldav.icloud.com:443

2020-03-24T21:19:14.990+0100	INFO	shadowsocks:  2020-03-24 21:19:14 INFO: new connection from client, 3 opened client connections

2020-03-24T21:19:14.991+0100	INFO	shadowsocks:  2020-03-24 21:19:14 INFO: [8388] connect to p38-caldav.icloud.com:443

2020-03-24T21:19:15.780+0100	INFO	shadowsocks:  2020-03-24 21:19:15 INFO: new connection from client, 4 opened client connections

2020-03-24T21:19:15.781+0100	INFO	shadowsocks:  2020-03-24 21:19:15 INFO: [8388] connect to ipinfo.io:443

2020-03-24T21:19:18.772+0100	INFO	shadowsocks:  2020-03-24 21:19:18 INFO: new connection from client, 5 opened client connections

2020-03-24T21:19:18.773+0100	INFO	shadowsocks:  2020-03-24 21:19:18 INFO: [8388] connect to api.ipify.org:443

2020-03-24T21:19:22.142+0100	INFO	shadowsocks:  2020-03-24 21:19:22 INFO: failed to lookup v4 address Timeout while contacting DNS servers

2020-03-24T21:19:22.142+0100	INFO	shadowsocks:  2020-03-24 21:19:22 INFO: failed to lookup v6 address Timeout while contacting DNS servers

2020-03-24T21:19:22.142+0100	INFO	shadowsocks:  2020-03-24 21:19:22 INFO: close a connection from client, 4 opened client connections

2020-03-24T21:19:23.532+0100	INFO	openvpn: Tue Mar 24 21:19:23 2020 [UNDEF] Inactivity timeout (--ping-restart), restarting

2020-03-24T21:19:23.532+0100	INFO	openvpn: Tue Mar 24 21:19:23 2020 SIGUSR1[soft,ping-restart] received, process restarting

2020-03-24T21:19:23.950+0100	INFO	shadowsocks:  2020-03-24 21:19:23 INFO: failed to lookup v4 address Timeout while contacting DNS servers

2020-03-24T21:19:24.743+0100	INFO	shadowsocks:  2020-03-24 21:19:24 INFO: failed to lookup v6 address Timeout while contacting DNS servers

2020-03-24T21:19:24.743+0100	INFO	shadowsocks:  2020-03-24 21:19:24 INFO: failed to lookup v4 address Timeout while contacting DNS servers

2020-03-24T21:19:24.743+0100	INFO	shadowsocks:  2020-03-24 21:19:24 INFO: close a connection from client, 3 opened client connections

2020-03-24T21:19:24.743+0100	INFO	shadowsocks:  2020-03-24 21:19:24 INFO: failed to lookup v4 address Timeout while contacting DNS servers

2020-03-24T21:19:24.744+0100	INFO	shadowsocks:  2020-03-24 21:19:24 INFO: failed to lookup v6 address Timeout while contacting DNS servers

2020-03-24T21:19:24.744+0100	INFO	shadowsocks:  2020-03-24 21:19:24 INFO: close a connection from client, 2 opened client connections

2020-03-24T21:19:24.744+0100	INFO	shadowsocks:  2020-03-24 21:19:24 INFO: failed to lookup v6 address Timeout while contacting DNS servers

2020-03-24T21:19:24.744+0100	INFO	shadowsocks:  2020-03-24 21:19:24 INFO: close a connection from client, 1 opened client connections

2020-03-24T21:19:24.901+0100	INFO	shadowsocks:  2020-03-24 21:19:24 INFO: new connection from client, 2 opened client connections

2020-03-24T21:19:24.901+0100	INFO	shadowsocks:  2020-03-24 21:19:24 INFO: [8388] connect to p28-caldav.icloud.com:443

2020-03-24T21:19:24.902+0100	INFO	shadowsocks:  2020-03-24 21:19:24 INFO: new connection from client, 3 opened client connections

2020-03-24T21:19:24.902+0100	INFO	shadowsocks:  2020-03-24 21:19:24 INFO: [8388] connect to p38-caldav.icloud.com:443

2020-03-24T21:19:24.903+0100	INFO	shadowsocks:  2020-03-24 21:19:24 INFO: failed to lookup v4 address Could not contact DNS servers

2020-03-24T21:19:27.736+0100	INFO	shadowsocks:  2020-03-24 21:19:27 INFO: failed to lookup v6 address Timeout while contacting DNS servers

2020-03-24T21:19:27.736+0100	INFO	shadowsocks:  2020-03-24 21:19:27 INFO: failed to lookup v4 address Timeout while contacting DNS servers

2020-03-24T21:19:27.737+0100	INFO	shadowsocks:  2020-03-24 21:19:27 INFO: close a connection from client, 2 opened client connections

2020-03-24T21:19:33.533+0100	INFO	openvpn: Tue Mar 24 21:19:33 2020 TCP/UDP: Preserving recently used remote address: [AF_INET]185.253.96.3:443

2020-03-24T21:19:33.533+0100	INFO	openvpn: Tue Mar 24 21:19:33 2020 UDP link local: (not bound)

2020-03-24T21:19:33.533+0100	INFO	openvpn: Tue Mar 24 21:19:33 2020 UDP link remote: [AF_INET]185.253.96.3:443

2020-03-24T21:19:35.444+0100	INFO	shadowsocks:  2020-03-24 21:19:35 INFO: failed to lookup v4 address Timeout while contacting DNS servers

2020-03-24T21:19:35.444+0100	INFO	shadowsocks:  2020-03-24 21:19:35 INFO: failed to lookup v6 address Timeout while contacting DNS servers

2020-03-24T21:19:35.444+0100	INFO	shadowsocks:  2020-03-24 21:19:35 INFO: close a connection from client, 1 opened client connections

2020-03-24T21:19:35.444+0100	INFO	shadowsocks:  2020-03-24 21:19:35 INFO: failed to lookup v6 address Timeout while contacting DNS servers

2020-03-24T21:19:35.444+0100	INFO	shadowsocks:  2020-03-24 21:19:35 INFO: close a connection from client, 0 opened client connections

2020-03-24T21:20:33.322+0100	INFO	openvpn: Tue Mar 24 21:20:33 2020 [UNDEF] Inactivity timeout (--ping-restart), restarting

2020-03-24T21:20:33.323+0100	INFO	openvpn: Tue Mar 24 21:20:33 2020 SIGUSR1[soft,ping-restart] received, process restarting

2020-03-24T21:20:43.322+0100	INFO	openvpn: Tue Mar 24 21:20:43 2020 TCP/UDP: Preserving recently used remote address: [AF_INET]46.166.143.98:443

2020-03-24T21:20:43.323+0100	INFO	openvpn: Tue Mar 24 21:20:43 2020 UDP link local: (not bound)

2020-03-24T21:20:43.323+0100	INFO	openvpn: Tue Mar 24 21:20:43 2020 UDP link remote: [AF_INET]46.166.143.98:443
```

**Comment by qdm12** [comment]
What version are you using (see top of logs)?

Also can you disable shadowsocks in order to clean out the logs? Thanks!

**Comment by Frepke** [comment]
> What version are you using (see top of logs)?
> 
> Also can you disable shadowsocks in order to clean out the logs? Thanks!

```      
===== Shadowsocks, Tinyproxy and Go =====

=========================================

=== Made with ❤  by github.com/qdm12 ====

=========================================


Running version windscribe built on 2020-03-24T18:07:34Z (commit a0513c0)


📣  Support for Windscribe


🔧  Need help? https://github.com/qdm12/private-internet-access-docker/issues/new

💻  Email? quentin.mcgaw@gmail.com

☕  Slack? Join from the Slack button on Github

💸  Help me? https://github.com/sponsors/qdm12

2020-03-24T21:26:36.966+0100	INFO	OpenVPN version: 2.4.8

2020-03-24T21:26:36.971+0100	INFO	Unbound version: 1.9.6

2020-03-24T21:26:36.974+0100	INFO	IPtables version: v1.8.3

2020-03-24T21:26:37.040+0100	INFO	TinyProxy version: 1.10.0

2020-03-24T21:26:37.044+0100	INFO	ShadowSocks version: 3.3.4

2020-03-24T21:26:37.045+0100	INFO	Settings summary below:

OpenVPN settings:

|--Network protocol: udp

|--Verbosity level: 1

|--Run as root: no

|--Target IP address: <nil>

Windscribe settings:

 |--User: [redacted]

 |--Password: [redacted]

 |--Region: netherlands

DNS over TLS settings:

 |--DNS over TLS provider:

  |--cloudflare

  |--quad9

 |--Caching: enabled

 |--Block malicious: enabled

 |--Block surveillance: enabled

 |--Block ads: enabled

 |--Allowed hostnames:

  |--

 |--Private addresses:

  |--127.0.0.1/8

  |--10.0.0.0/8

  |--172.16.0.0/12

  |--192.168.0.0/16

  |--169.254.0.0/16

  |--::1/128

  |--fc00::/7

  |--fe80::/10

  |--::ffff:0:0/96

 |--Verbosity level: 1/5

 |--Verbosity details level: 0/4

 |--Validation log level: 0/2

 |--IPv6 resolution: disabed

Firewall settings:

 |--Allowed subnets: 10.54.1.0/24

TinyProxy settings: disabled

ShadowSocks settings: disabled


2020-03-24T21:26:37.045+0100	INFO	openvpn configurator: checking for device /dev/net/tun

2020-03-24T21:26:37.046+0100	WARN	TUN device is not available: open /dev/net/tun: no such file or directory

2020-03-24T21:26:37.046+0100	INFO	openvpn configurator: creating /dev/net/tun

2020-03-24T21:26:37.046+0100	INFO	openvpn configurator: writing auth file /etc/openvpn/auth.conf

2020-03-24T21:26:37.047+0100	INFO	firewall configurator: accepting all traffic

2020-03-24T21:26:37.054+0100	INFO	dns configurator: using DNS address 1.1.1.1 internally

2020-03-24T21:26:37.054+0100	INFO	dns configurator: downloading root hints from https://raw.githubusercontent.com/qdm12/files/master/named.root.updated

2020-03-24T21:26:37.054+0100	INFO	Launching standard output merger

2020-03-24T21:26:37.453+0100	INFO	dns configurator: downloading root key from https://raw.githubusercontent.com/qdm12/files/master/root.key.updated

2020-03-24T21:26:37.629+0100	INFO	dns configurator: generating Unbound configuration

2020-03-24T21:26:38.638+0100	INFO	dns configurator: 131467 hostnames blocked overall

2020-03-24T21:26:38.638+0100	INFO	dns configurator: 9 IP addresses blocked overall

2020-03-24T21:26:38.966+0100	INFO	dns configurator: starting unbound

2020-03-24T21:26:38.968+0100	INFO	dns configurator: using DNS address 127.0.0.1 internally

2020-03-24T21:26:38.968+0100	INFO	dns configurator: using DNS address 127.0.0.1 system wide

2020-03-24T21:26:38.973+0100	WARN	could not resolve github.com (try 1 of 10)

2020-03-24T21:26:40.431+0100	INFO	unbound: [1585081600] unbound[21:0] notice: init module 0: validator

2020-03-24T21:26:40.431+0100	INFO	unbound: [1585081600] unbound[21:0] notice: init module 1: iterator

2020-03-24T21:26:40.467+0100	INFO	unbound: [1585081600] unbound[21:0] info: start of service (unbound 1.9.6).

2020-03-24T21:26:40.513+0100	INFO	unbound: [1585081600] unbound[21:0] info: generate keytag query _ta-4a5c-4f66. NULL IN

2020-03-24T21:26:40.727+0100	INFO	firewall configurator: detecting default network route

2020-03-24T21:26:40.728+0100	INFO	firewall configurator: default route found: interface eth0, gateway 172.17.0.1, subnet 172.17.0.0/16

2020-03-24T21:26:40.730+0100	INFO	firewall configurator: adding 10.54.1.0/24 as route via eth0

2020-03-24T21:26:40.732+0100	INFO	firewall configurator: clearing all rules

2020-03-24T21:26:40.740+0100	INFO	firewall configurator: blocking all traffic

2020-03-24T21:26:40.753+0100	INFO	firewall configurator: creating general rules

2020-03-24T21:26:40.766+0100	INFO	firewall configurator: allowing output traffic to VPN server 72.11.157.67 through eth0 on port udp 443

2020-03-24T21:26:40.770+0100	INFO	firewall configurator: allowing output traffic to VPN server 84.17.46.2 through eth0 on port udp 443

2020-03-24T21:26:40.773+0100	INFO	firewall configurator: allowing output traffic to VPN server 185.253.96.3 through eth0 on port udp 443

2020-03-24T21:26:40.777+0100	INFO	firewall configurator: allowing output traffic to VPN server 37.120.192.19 through eth0 on port udp 443

2020-03-24T21:26:40.779+0100	INFO	firewall configurator: allowing output traffic to VPN server 185.212.171.131 through eth0 on port udp 443

2020-03-24T21:26:40.782+0100	INFO	firewall configurator: allowing output traffic to VPN server 46.166.143.98 through eth0 on port udp 443

2020-03-24T21:26:40.785+0100	INFO	firewall configurator: allowing output traffic to VPN server 72.11.157.35 through eth0 on port udp 443

2020-03-24T21:26:40.790+0100	INFO	firewall configurator: accepting input and output traffic for 172.17.0.0/16

2020-03-24T21:26:40.795+0100	INFO	firewall configurator: accepting input traffic through eth0 from 10.54.1.0/24 to 172.17.0.0/16

2020-03-24T21:26:40.797+0100	INFO	firewall configurator: accepting output traffic through eth0 from 172.17.0.0/16 to 10.54.1.0/24

2020-03-24T21:26:40.799+0100	INFO	openvpn configurator: starting openvpn

2020-03-24T21:26:40.803+0100	INFO	openvpn: Tue Mar 24 21:26:40 2020 OpenVPN 2.4.8 x86_64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on Feb  7 2020

2020-03-24T21:26:40.803+0100	INFO	openvpn: Tue Mar 24 21:26:40 2020 library versions: OpenSSL 1.1.1d  10 Sep 2019, LZO 2.10

2020-03-24T21:26:40.810+0100	INFO	openvpn: Tue Mar 24 21:26:40 2020 TCP/UDP: Preserving recently used remote address: [AF_INET]72.11.157.67:443

2020-03-24T21:26:40.810+0100	INFO	openvpn: Tue Mar 24 21:26:40 2020 UDP link local: (not bound)

2020-03-24T21:26:40.810+0100	INFO	openvpn: Tue Mar 24 21:26:40 2020 UDP link remote: [AF_INET]72.11.157.67:443

2020-03-24T21:26:40.810+0100	INFO	openvpn: Tue Mar 24 21:26:40 2020 NOTE: UID/GID downgrade will be delayed because of --client, --pull, or --up-delay```

**Comment by qdm12** [comment]
Also, what region are you using? Maybe try with another region? Can anyone else subscribed with Windscribe also try?

And does it work using one of the openvpn files using openvpn on your laptop/desktop?

Sorry, I'm a bit running out of options on that one 😕

**Comment by Frepke** [comment]
Maybe the problem is me, trying it with a free Windscribe accound.

**Comment by Frepke** [comment]
I build a Windscribe-plan for two country's. Now I can get in the sites PRO-section.
I've to genereate my own .ovpn-file by selecting some inputfields:
![afbeelding](https://user-images.githubusercontent.com/6153559/77518289-d151fc80-6e7d-11ea-888a-7e3f65a96169.png)

- Location: Netherlands*
- UDP or TCP = UDP*
- Port 443, 80, 53, 123, 1194 or 54783 = 443*
- Cipher AES-CBC or AES-GCM = AES-CBC*
* = my selection
My selection generates a .ovpn-file (see below) and comes with specific credentials.
I hope this helps.

kr.,
Frepke

**Windscribe-Netherlands.ovpn**
```client
dev tun
proto udp
remote nl.windscribe.com 443

nobind
auth-user-pass

resolv-retry infinite

auth SHA512
cipher AES-256-CBC
comp-lzo
verb 2
mute-replay-warnings
remote-cert-tls server
persist-key
persist-tun

key-direction 1
<ca>
-----BEGIN CERTIFICATE-----
MIIF3DCCA8SgAwIBAgIJAMsOivWTmu9fMA0GCSqGSIb3DQEBCwUAMHsxCzAJBgNV
BAYTAkNBMQswCQYDVQQIDAJPTjEQMA4GA1UEBwwHVG9yb250bzEbMBkGA1UECgwS
V2luZHNjcmliZSBMaW1pdGVkMRMwEQYDVQQLDApPcGVyYXRpb25zMRswGQYDVQQD
DBJXaW5kc2NyaWJlIE5vZGUgQ0EwHhcNMTYwMzA5MDMyNjIwWhcNNDAxMDI5MDMy
NjIwWjB7MQswCQYDVQQGEwJDQTELMAkGA1UECAwCT04xEDAOBgNVBAcMB1Rvcm9u
dG8xGzAZBgNVBAoMEldpbmRzY3JpYmUgTGltaXRlZDETMBEGA1UECwwKT3BlcmF0
aW9uczEbMBkGA1UEAwwSV2luZHNjcmliZSBOb2RlIENBMIICIjANBgkqhkiG9w0B
AQEFAAOCAg8AMIICCgKCAgEAruBtLR1Vufd71LeQEqChgHS4AQJ0fSRner0gmZPE
r2TL5uWboOEWXFFoEUTthF+P/N8yy3xRZ8HhG/zKlmJ1xw+7KZRbTADD6shJPj3/
uvTIO80sU+9LmsyKSWuPhQ1NkgNA7rrMTfz9eHJ2MVDs4XCpYWyX9iuAQrHSY6aP
q+4TpCbUgprkM3Gwjh9RSt9IoDoc4CF2bWSaVepUcL9yz/SXLPzFx2OT9rFrDhL3
ryHRzJQ/tA+VD8A7lo8bhOcDqiXgEFmVOZNMLw+r167Qq1Ck7X86yr2mnW/6HK2g
JOvY0/SPKukfGJAiYZKdG+fe4ekyYcAVhDfPJg7rF9wUqPwUzejJyAs1K18JwX94
Y8fnD6vQobjpC3qfHtwQP7Uj2AcI6QC8ytWDegV6UIkHXAMXBQSX5suSQoE11deG
32cy7nyp5vhgy31rTyNoopqlcCAhPm6k0jVVQbvXhLcpTSL8iCCoMdrP28i/xsfv
ktBAkl5giHMdK6hxqWgPI+Bx9uPIhRp3fJ2z8AgFm8g1ARB2ZzQ+OZZ2RUIkJuUK
hi2kUhgKSAQ+eF89aoqDjp/J1miZqGRzt4DovSZfQOeL01RkKHEibAPYCfgHG2ZS
woLoeaxE2vNZiX4dpXiOQYTOIXOwEPZzPvfTQf9T4Kxvx3jzQnt3PzjlMCqKk3Ai
pm8CAwEAAaNjMGEwHQYDVR0OBBYEFEH2v9F2z938Ebngsj9RkVSSgs45MB8GA1Ud
IwQYMBaAFEH2v9F2z938Ebngsj9RkVSSgs45MA8GA1UdEwEB/wQFMAMBAf8wDgYD
VR0PAQH/BAQDAgGGMA0GCSqGSIb3DQEBCwUAA4ICAQAgI6NgYkVo5rB6yKStgHjj
ZsINsgEvoMuHwkM0YaV22XtKNiHdsiOmY/PGCRemFobTEHk5XHcvcOTWv/D1qVf8
fI21WAoNQVH7h8KEsr4uMGKCB6Lu8l6xALXRMjo1xb6JKBWXwIAzUu691rUD2exT
1E+A5t+xw+gzqV8rWTMIoUaH7O1EKjN6ryGW71Khiik8/ETrP3YT32ZbS2P902iM
Kw9rpmuS0wWhnO5k/iO/6YNA1ZMV5JG5oZvZQYEDk7enLD9HvqazofMuy/Sz/n62
ZCDdQsnabzxl04wwv5Y3JZbV/6bOM520GgdJEoDxviY05ax2Mz05otyBzrAVjFw9
RZt/Ls8ATifu9BusZ2ootvscdIuE3x+ZCl5lvANcFEnvgGw0qpCeASLpsfxwq1dR
gIn7BOiTauFv4eoeFAQvCD+l+EKGWKu3M2y19DgYX94N2+Xs2bwChroaO5e4iFem
MLMuWKZvYgnqS9OAtRSYWbNX/wliiPz7u13yj+qSWgMfu8WPYNQlMZJXuGWUvKLE
XCUExlu7/o8D4HpsVs30E0pUdaqN0vExB1KegxPWWrmLcYnPG3knXpkC3ZBZ5P/e
l/2eyhZRy9ydiITF8gM3L08E8aeqvzZMw2FDSmousydIzlXgeS5VuEf+lUFA2h8o
ZYGQgrLt+ot8MbLhJlkp4Q==
-----END CERTIFICATE-----
</ca>
<tls-auth>
-----BEGIN OpenVPN Static key V1-----
5801926a57ac2ce27e3dfd1dd6ef8204
2d82bd4f3f0021296f57734f6f1ea714
a6623845541c4b0c3dea0a050fe6746c
b66dfab14cda27e5ae09d7c155aa554f
399fa4a863f0e8c1af787e5c602a801d
3a2ec41e395a978d56729457fe6102d7
d9e9119aa83643210b33c678f9d4109e
3154ac9c759e490cb309b319cf708cae
83ddadc3060a7a26564d1a24411cd552
fe6620ea16b755697a4fc5e6e9d0cfc0
c5c4a1874685429046a424c026db672e
4c2c492898052ba59128d46200b40f88
0027a8b6610a4d559bdc9346d33a0a6b
08e75c7fd43192b162bfd0aef0c716b3
1584827693f676f9a5047123466f0654
eade34972586b31c6ce7e395f4b478cb
-----END OpenVPN Static key V1-----
</tls-auth>```

**Comment by Frepke** [comment]
Quentin,

Is it possible to modify the .ovpn file in the dockercontainer by myselve and restart the service afterwards in the container?

**Comment by qdm12** [comment]
I think `key-direction 1` in the code, I'll add it in my lunch break.

You could docker run the container with `--entrypoint=/bin/sh`, then run `/entrypoint`, ctrl-c to exit, modify the config file generated (in /etc/openvpn/) and run `openvpn` to test. That would be a good way to debug what's wrong. But a bit tedious unfortunately.

**Comment by Frepke** [comment]
> 
> 
> I think `key-direction 1` in the code, I'll add it in my lunch break.
> 
> You could docker run the container with `--entrypoint=/bin/sh`, then run `/entrypoint`, ctrl-c to exit, modify the config file generated (in /etc/openvpn/) and run `openvpn` to test. That would be a good way to debug what's wrong. But a bit tedious unfortunately.

Tried it in the Portainer-console.
vi /etc/openvpn/target.ovpn and add key-direction 1
saved the file and then openvpn /etc/openvpn/target.ovpn 

still not running :(

**Comment by qdm12** [comment]
I added the missing `key-direction 1`. I think you need to run `openvpn ---config /etc/openvpn/target.ovpn` instead.

Also I need to add the port selection option for Windscribe, I wasn't aware you could pick it. The encryption one will probably come with #94 

Do Windscribe have a free plan? I will try with it if so.

**Comment by Frepke** [comment]
Sorry, still the same errors:
```
2020-03-25T19:40:37.318+0100	INFO	openvpn: Wed Mar 25 19:40:37 2020 UDP link local: (not bound)

2020-03-25T19:40:37.318+0100	INFO	openvpn: Wed Mar 25 19:40:37 2020 UDP link remote: [AF_INET]46.166.143.98:443

2020-03-25T19:41:37.639+0100	INFO	openvpn: Wed Mar 25 19:41:37 2020 [UNDEF] Inactivity timeout (--ping-restart), restarting

2020-03-25T19:41:37.639+0100	INFO	openvpn: Wed Mar 25 19:41:37 2020 SIGUSR1[soft,ping-restart] received, process restarting

2020-03-25T19:41:47.640+0100	INFO	openvpn: Wed Mar 25 19:41:47 2020 TCP/UDP: Preserving recently used remote address: [AF_INET]72.11.157.35:443

2020-03-25T19:41:47.640+0100	INFO	openvpn: Wed Mar 25 19:41:47 2020 UDP link local: (not bound)

2020-03-25T19:41:47.640+0100	INFO	openvpn: Wed Mar 25 19:41:47 2020 UDP link remote: [AF_INET]72.11.157.35:443```

**Comment by Frepke** [comment]
> I added the missing `key-direction 1`. I think you need to run `openvpn ---config /etc/openvpn/target.ovpn` instead.
> 
> Also I need to add the port selection option for Windscribe, I wasn't aware you could pick it. The encryption one will probably come with #94
> 
> Do Windscribe have a free plan? I will try with it if so.

Yes, they have a free plan for 10GB a month on selected servers.

**Comment by qdm12** [comment]
@Frepke I have signed up, paid them $2 (probably to save hours of debugging 😆 ) . The bug was me copy pasta some lines of code, and basically it was using the mullvad configurator instead of the newer windscribe, hence making the ovpn file not the right one! It's fixed now, I tested it and it connects successfully 👍 Let me know how it goes for you and we can then finally close this long issue! 🎉

**Comment by Frepke** [comment]
I've tried my two subscribed regions with Windscribe (Netherlands and United Kingdom). Both connected instantly and without errors 👍
And now I see the correct target.ovpn in /etc/openvpn/ 😄 

I'm happy with this, thanks.

**Comment by Frepke** [comment]
Are you planning support for other providers?
Just a question, I've nothing in mind myself.

**Comment by elmerfdz** [comment]
Windscribe is working for me as well, nice one @qdm12 👏

Are there any plans to support port forwarding on Mullvad & Windscribe? 😄

**Comment by qdm12** [comment]
Great, thanks for the help guys!

I still need to support other ports for Windscribe before closing this issue.

Supporting other ciphers and auth algorithm for Windscribe and pia will come in #94 

I'll create another issue for port forwarding for Mullvad and Windscribe (didn't know you could!)

**Comment by qdm12** [comment]
Ok I've added support for custom ciphers and custom port, so please try it one last time and I'll merge it with the latest.

**Comment by Frepke** [comment]
Quentin,

I tested some different ports/protocols and all connect perfectly.
But there are some warnings in the log:
```
2020-03-27T07:56:18.079+0100	INFO	openvpn: Fri Mar 27 07:56:18 2020 Initialization Sequence Completed


2020-03-27T08:56:16.116+0100	INFO	openvpn: Fri Mar 27 08:56:16 2020 WARNING: 'link-mtu' is used inconsistently, local='link-mtu 1602', remote='link-mtu 1550'


2020-03-27T08:56:16.116+0100	INFO	openvpn: Fri Mar 27 08:56:16 2020 WARNING: 'cipher' is used inconsistently, local='cipher AES-256-CBC', remote='cipher AES-256-GCM'


2020-03-27T08:56:16.116+0100	INFO	openvpn: Fri Mar 27 08:56:16 2020 WARNING: 'auth' is used inconsistently, local='auth SHA512', remote='auth [null-digest]'


2020-03-27T09:56:16.893+0100	INFO	openvpn: Fri Mar 27 09:56:16 2020 WARNING: 'link-mtu' is used inconsistently, local='link-mtu 1602', remote='link-mtu 1550'


2020-03-27T09:56:16.893+0100	INFO	openvpn: Fri Mar 27 09:56:16 2020 WARNING: 'cipher' is used inconsistently, local='cipher AES-256-CBC', remote='cipher AES-256-GCM'


2020-03-27T09:56:16.893+0100	INFO	openvpn: Fri Mar 27 09:56:16 2020 WARNING: 'auth' is used inconsistently, local='auth SHA512', remote='auth [null-digest]'```

**Comment by qdm12** [comment]
@Frepke It seems to be related to [this Reddit post](https://www.reddit.com/r/Windscribe/comments/9jxezv/openvpn_desktop_client_not_windscribe_client/) because their server has a different configuration than the client config they give (which is silly but anyway, pia does the same...).

You can safely ignore these, but the client will essentially use the remote settings pushed by the server (i.e. even if you set encryption as AES-256-CBC it will use the GCM variant).

**Comment by Frepke** [comment]
Okay,

And thanks for this amazing piece of work!

**Comment by Frepke** [comment]
> Quentin,
> 
> I tested some different ports/protocols and all connect perfectly.
> But there are some warnings in the log:
> 
> ```
> 2020-03-27T07:56:18.079+0100	INFO	openvpn: Fri Mar 27 07:56:18 2020 Initialization Sequence Completed
> 2020-03-27T08:56:16.116+0100	INFO	openvpn: Fri Mar 27 08:56:16 2020 WARNING: 'link-mtu' is used inconsistently, local='link-mtu 1602', remote='link-mtu 1550'
> 2020-03-27T08:56:16.116+0100	INFO	openvpn: Fri Mar 27 08:56:16 2020 WARNING: 'cipher' is used inconsistently, local='cipher AES-256-CBC', remote='cipher AES-256-GCM'
> 2020-03-27T08:56:16.116+0100	INFO	openvpn: Fri Mar 27 08:56:16 2020 WARNING: 'auth' is used inconsistently, local='auth SHA512', remote='auth [null-digest]'
> 2020-03-27T09:56:16.893+0100	INFO	openvpn: Fri Mar 27 09:56:16 2020 WARNING: 'link-mtu' is used inconsistently, local='link-mtu 1602', remote='link-mtu 1550'
> 2020-03-27T09:56:16.893+0100	INFO	openvpn: Fri Mar 27 09:56:16 2020 WARNING: 'cipher' is used inconsistently, local='cipher AES-256-CBC', remote='cipher AES-256-GCM'
> 2020-03-27T09:56:16.893+0100	INFO	openvpn: Fri Mar 27 09:56:16 2020 WARNING: 'auth' is used inconsistently, local='auth SHA512', remote='auth [null-digest]
> ```

I've changed my settings to ```OPENVPN_CIPHER=AES-256-GCM``` and all of the previous warnings are gone.
But now I've some strange INFO in the logfile:
```2020-03-27T15:53:45.948+0100	INFO	openvpn: Fri Mar 27 15:53:45 2020 Initialization Sequence Completed
2020-03-27T17:41:58.877+0100	INFO	unbound: [1585327318] unbound[21:0] info: generate keytag query _ta-4a5c-4f66. NULL IN
2020-03-27T18:22:27.379+0100	INFO	openvpn: Fri Mar 27 18:22:27 2020 TLS Error: local/remote TLS keys are out of sync: [AF_INET]185.212.171.131:54783 [3]
2020-03-27T18:32:36.106+0100	INFO	openvpn: Fri Mar 27 18:32:36 2020 TLS Error: local/remote TLS keys are out of sync: [AF_INET]185.212.171.131:54783 [3]
2020-03-27T19:35:33.026+0100	INFO	openvpn: Fri Mar 27 19:35:33 2020 TLS Error: local/remote TLS keys are out of sync: [AF_INET]185.212.171.131:54783 [0]
2020-03-27T20:14:21.200+0100	INFO	unbound: [1585336461] unbound[21:0] info: generate keytag query _ta-4a5c-4f66. NULL IN
2020-03-27T20:43:09.765+0100	INFO	openvpn: Fri Mar 27 20:43:09 2020 TLS Error: local/remote TLS keys are out of sync: [AF_INET]185.212.171.131:54783 [1]
2020-03-27T20:57:01.129+0100	INFO	openvpn: Fri Mar 27 20:57:01 2020 TLS Error: local/remote TLS keys are out of sync: [AF_INET]185.212.171.131:54783 [6]
2020-03-27T21:45:10.359+0100	INFO	openvpn: Fri Mar 27 21:45:10 2020 TLS Error: local/remote TLS keys are out of sync: [AF_INET]185.212.171.131:54783 [2]
2020-03-27T22:38:08.346+0100	INFO	openvpn: Fri Mar 27 22:38:08 2020 TLS Error: local/remote TLS keys are out of sync: [AF_INET]185.212.171.131:54783 [0]
2020-03-27T22:53:58.590+0100	INFO	unbound: [1585346038] unbound[21:0] info: generate keytag query _ta-4a5c-4f66. NULL IN
2020-03-28T00:11:05.895+0100	INFO	openvpn: Sat Mar 28 00:11:05 2020 TLS Error: local/remote TLS keys are out of sync: [AF_INET]185.212.171.131:54783 [2]
2020-03-28T01:53:41.956+0100	INFO	unbound: [1585356821] unbound[21:0] info: generate keytag query _ta-4a5c-4f66. NULL IN
2020-03-28T02:28:53.615+0100	INFO	openvpn: Sat Mar 28 02:28:53 2020 TLS Error: local/remote TLS keys are out of sync: [AF_INET]185.212.171.131:54783 [0]
2020-03-28T02:50:47.885+0100	INFO	openvpn: Sat Mar 28 02:50:47 2020 AEAD Decrypt error: cipher final failed
2020-03-28T03:05:02.521+0100	INFO	openvpn: Sat Mar 28 03:05:02 2020 TLS Error: local/remote TLS keys are out of sync: [AF_INET]185.212.171.131:54783 [0]
2020-03-28T03:24:30.969+0100	INFO	openvpn: Sat Mar 28 03:24:30 2020 TLS Error: local/remote TLS keys are out of sync: [AF_INET]185.212.171.131:54783 [0]
2020-03-28T04:07:17.280+0100	INFO	unbound: [1585364837] unbound[21:0] info: generate keytag query _ta-4a5c-4f66. NULL IN
2020-03-28T06:15:21.946+0100	INFO	openvpn: Sat Mar 28 06:15:21 2020 TLS Error: local/remote TLS keys are out of sync: [AF_INET]185.212.171.131:54783 [0]
2020-03-28T06:26:09.106+0100	INFO	openvpn: Sat Mar 28 06:26:09 2020 TLS Error: local/remote TLS keys are out of sync: [AF_INET]185.212.171.131:54783 [1]
2020-03-28T07:02:07.489+0100	INFO	openvpn: Sat Mar 28 07:02:07 2020 TLS Error: local/remote TLS keys are out of sync: [AF_INET]185.212.171.131:54783 [2]
2020-03-28T07:20:09.773+0100	INFO	unbound: [1585376409] unbound[21:0] info: generate keytag query _ta-4a5c-4f66. NULL IN
2020-03-28T07:23:21.761+0100	INFO	openvpn: Sat Mar 28 07:23:21 2020 TLS Error: local/remote TLS keys are out of sync: [AF_INET]185.212.171.131:54783 [2]
2020-03-28T08:51:23.704+0100	INFO	unbound: [1585381883] unbound[21:0] info: generate keytag query _ta-4a5c-4f66. NULL IN
2020-03-28T09:31:27.803+0100	INFO	openvpn: Sat Mar 28 09:31:27 2020 TLS Error: local/remote TLS keys are out of sync: [AF_INET]185.212.171.131:54783 [0]
2020-03-28T10:16:51.155+0100	INFO	unbound: [1585387011] unbound[21:0] info: generate keytag query _ta-4a5c-4f66. NULL IN
```

**Comment by qdm12** [comment]
I tried it on my machine, it seems to work without errors nor warnings using `AES-256-GCM`. Have you set the timezone `TZ` environment variable? I read that it could be related to a bad system clock time.

**Comment by Frepke** [comment]
> I tried it on my machine, it seems to work without errors nor warnings using `AES-256-GCM`. Have you set the timezone `TZ` environment variable? I read that it could be related to a bad system clock time.

Yes, TZ is set.

I missed a line in the logfile I see:
```2020-03-27T15:53:45.948+0100	INFO	openvpn: Fri Mar 27 15:53:45 2020 Initialization Sequence Completed```
Now you can see the INFO messages start after approx 2 hours uptime

**Comment by qdm12** [comment]
I will now merge windscribe into the latest Docker image and close the issue. See #113  for the bug you found and #112 for finishing renaming the repository and Docker image.

@Frepke  I do have the AHEAD Decrypt error, but not `TLS Error: local/remote TLS keys are out of sync`. I think that's most likely related to your time setting. What does

```sh
docker exec -it <container-name> date
```

give you?

**Comment by Frepke** [comment]
> I will now merge windscribe into the latest Docker image and close the issue. See #113 for the bug you found and #112 for finishing renaming the repository and Docker image.
> 
> @Frepke I do have the AHEAD Decrypt error, but not `TLS Error: local/remote TLS keys are out of sync`. I think that's most likely related to your time setting. What does
> 
> ```shell
> docker exec -it <container-name> date
> ```
> 
> give you?

```
docker exec -it vpn date
Sun Mar 29 20:30:21 CEST 2020

**Comment by qdm12** [comment]
Uh that's strange, I get `Sun Mar 29 19:04:34 UTC 2020` (so UTC) although my `TZ=America/Montreal`. Does `docker exec -it vpn echo $TZ` give you your timezone? Do you mind creating (yet another) issue for that particular problem? Thanks!
